napalm/eos/eos.py

```diff
@@ -131,60 +136,120 @@ def _process_optional_args(self, optional_args):
             "transport", optional_args.get("eos_transport", "https")
         )
         self.fn0039_config = optional_args.pop("eos_fn0039_config", False)
-        try:
-            self.transport_class = pyeapi.client.TRANSPORTS[transport]
-        except KeyError:
-            raise ConnectionException("Unknown transport: {}".format(self.transport))
+        self.transport = transport
+        if transport == "ssh":
+            self.transport_class = "ssh"
```

Comment:
transport_class as a string runs into an issue on line 150 and again in line 155 where a string only has self, so no optional args get passed into netmiko_optional_args.

Reply:
I've merged the changes to the napalm/eos/eos.py file.